server/lobbies.js

```diff
@@ -15,6 +18,20 @@ export class Lobby {
     // TODO temp disable
     // eslint-disable-next-line no-use-before-define
     save(lobbies);
+    this.hit();
+  }
+
+  // every time this function is called, a new timeout is generated
+  hit() {
+    this.expirationId += 1;
+    const id = this.expirationId;
+    setTimeout(() => this.expire(id), LOBBY_LIFESPAN);
```

Comment:
I think it would be better to use `timeoutId` returned by `setTimeout` as `expirationId` and then call `clearTimeout(timeoutId)` at the start of `hit` method.

Reply:
Hmm cool, I didn't know it returned an id